### PR TITLE
Plugin updates

### LINKED ISSUE #7471: Jetpack Plugin Support
We want to add Jetpack Plugin Support to the apps with the following functionality:

- Browse Plugins
- Activate/Deactivate a Plugin
- Turn on/off autoupdate for a Plugin
- Update a Plugin

### PR DESCRIPTION
**Pull request by koke**
Adds support to update a plugin.

plugin-updates

wpios-plugin-updates

Fixes #7471

To test:

- Make sure you have updates available for a plugin
- Go to the plugins list, the update icon should show
- Go to the plugin detail, it should say what version is available
- Tap the update button, the button should say "Updating", and if you go back to the list, the update icon should animate.
- When the update finishes, the update icon in the list should disappear, and the "version available" row should be gone from the plugin detail view.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Plugins/PluginListRow.swift [hunk 1]
```diff
@@ -15,6 +18,24 @@ struct PluginListRow: ImmuTableRow {
         let iconPlaceholder = Gridicon.iconOfType(.plugins, withSize: iconSize)
         cell.imageView?.downloadResizedImage(iconURL, placeholderImage: iconPlaceholder, pointSize: iconSize)
         cell.selectionStyle = .default
-        cell.accessoryType = .disclosureIndicator
+        switch updateState {
+        case .available(_):
+            cell.accessoryView = UIImageView(image: #imageLiteral(resourceName: "gridicon-sync-circled"))
+        case .updating(_):
+            cell.accessoryView = spinningImageView(image: #imageLiteral(resourceName: "gridicon-sync-circled"))
+        case .updated:
```
[the remark this answers is not in the document]
Comment (koke) [reply WordPress/Classes/ViewRelated/Plugins/PluginListRow.swift]: TIL 😮

File: WordPressKit/WordPressKit/PluginServiceRemote.swift [hunk 2]
```diff
@@ -30,35 +30,63 @@ public class PluginServiceRemote: ServiceRemoteWordPressComREST {
...
+            return
+        }
+        let endpoint = "sites/\(siteID)/plugins/\(escapedPluginID)/update"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._1_2)!
+        let parameters = [String: AnyObject]()
+
+        wordPressComRestApi.POST(
+            path,
+            parameters: parameters,
+            success: { (responseObject,_)  in
```
Comment (elibud) [inline WordPressKit/WordPressKit/PluginServiceRemote.swift]: Missing space after the `,`

File: WordPress/Classes/Stores/PluginStore.swift [hunk 3]
```diff
@@ -56,6 +57,19 @@ struct PluginStoreState {
     var lastFetch = [JetpackSiteRef: Date]()
     var directoryEntries = [String: PluginDirectoryEntryState]()
     var fetchingDirectoryEntry = [String: Bool]()
+    var updatesInProgress = [JetpackSiteRef: Set<String>]()
```
[the remark this answers is not in the document]
Comment (koke) [reply WordPress/Classes/Stores/PluginStore.swift]: I don't think so. We're not returning these directly, but using them to map what we fetch from the server and avoid duplicate updates. If a plugin is removed, the value here will just not be used until the request finishes.

I think it's more of a list of ongoing update requests, so if we wanted to clear it, we should also cancel the request, which maybe we could do. I think it's a bit unlikely that you'd remove a plugin right after you tap update though, so probably not worth the extra complexity.

[the remark this answers is not in the document]
Comment (elibud) [reply WordPress/Classes/Stores/PluginStore.swift]: Understood and agreed 👍

**Comment by elibud** [comment]
Reporting back: I was able to successfully update from Jetpack 5.5 to Jetpack 5.6 on @astralbodies test site without an issue, so I think the glitch is related to updating from a beta version so I'm not concerned.

👏 looks and feels and works amazing, great job!

**Comment by koke** [comment]
🎉 Thanks for the review! One step closer to 🚢

**Pull request merged**